src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -243,14 +243,23 @@ public void DirectoryLongerThanMaxLongPath_ThrowsPathTooLongException()
 
         [ConditionalFact(nameof(LongPathsAreNotBlocked), nameof(UsingNewNormalization))]
         [ActiveIssue(20117, TargetFrameworkMonikers.Uap)]
-        [PlatformSpecific(TestPlatforms.Windows)]  // long directory path with extended syntax throws DirectoryNotFoundException
-        public void DirectoryLongerThanMaxLongPathWithExtendedSyntax_ThrowsDirectoryNotFoundException()
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void DirectoryLongerThanMaxLongPathWithExtendedSyntax_ThrowsException()
```

Comment:
Looking into it a bit further, it seems that both paths are exactly the same length (32,768 characters), but result in different exceptions being throw. See here: https://gist.github.com/steji113/bf755ba0ebad21e01b61b27a216c9fc1